Automattic/wp-calypso
# Checklist: Send to the template selector for Update your homepage item

As described in #37429, we want to send folks to the template picker when they are completing the `Update your homepage` checklist item. This accomplishes that goal.

#### Changes proposed in this Pull Request

* Add an ab test called `checklistUpdateHomepage`
* When selected for the above, update the URL for the item to create a new page (triggering the template picker) & add a query argument so the editor code knows to promote the page to the front
* Look for the query arg in the controller & pass it along to the iframe wrapper
* Update the code that runs in the editor iframe to subscribe to post status changes & notify the parent window when they occur
* Watch for post status changes in the parent & promote the page to the front when appropriate

|Before (Edits existing page)|After (Creates new page from template)|
|---|---|
|![update-homepage-before](https://user-images.githubusercontent.com/1587282/69058405-9a586f80-09e1-11ea-8679-2e22d192997d.gif)|![update-homepage-after](https://user-images.githubusercontent.com/1587282/69058433-a5ab9b00-09e1-11ea-8429-5985f3bbf042.gif)|


#### Testing instructions



* Apply D35426-code
* Sandbox the REST API
* Sandbox `widgets.wp.com`
* Sandbox your test site's `.wordpress.com` domain
* Visit `/home` (select your test site if shown the site picker)
* Click the CTA for the `Update your homepage` checklist item
  * You should be taken to the template picker for a new page
  * You should see a query arg in the address bar `new-homepage`
* Select a template & click the CTA button
* Make some edits
* Click the `Publish` CTA
  * The new page should be saved without error
* Browse to the root of the domain
* The new page should be shown
